torch/csrc/autograd/saved_variable.cpp

```diff
@@ -78,11 +78,15 @@ Variable SavedVariable::unpack(std::shared_ptr<Function> saved_for) const {
   // NB: saved views are unpacked as normal Variables (not views) even though
   // they still share the same storage. This works only because we never call
   // in-place functions on unpacked variables.
-  Variable var;
+  Variable var = as_variable_ref(data_).variable_data();
```

Comment:
It seems unclear whether we have invariant that the SavedVariable will only be unpacked once. (I tried removing detach/variable_data here and all tests pass.) If we do have that invariant, we should always use detach/variable_data here, so that the unpacked Variable is always a shallow copy of the SavedVariable.